pallets/ema-oracle/src/lib.rs

```diff
@@ -295,15 +304,39 @@ pub mod pallet {
 		}
 
 		#[pallet::call_index(2)]
-		#[pallet::weight(<T as Config>::WeightInfo::update_bifrost_oracle())]
+		#[pallet::weight(<T as Config>::WeightInfo::update_bifrost_oracle())] //TODO: rebenchmark
 		pub fn update_bifrost_oracle(
 			origin: OriginFor<T>,
 			//NOTE: these must be boxed becasue of https://github.com/paritytech/polkadot-sdk/blob/6875d36b2dba537f3254aad3db76ac7aa656b7ab/substrate/frame/utility/src/lib.rs#L150
-			_asset_a: Box<polkadot_xcm::VersionedLocation>,
-			_asset_b: Box<polkadot_xcm::VersionedLocation>,
-			_price: (Balance, Balance),
+			asset_a: Box<polkadot_xcm::VersionedLocation>,
+			asset_b: Box<polkadot_xcm::VersionedLocation>,
+			price: (Balance, Balance),
 		) -> DispatchResult {
+			//TODO: add integration test due to mocked out currencyidcovnert
 			T::BifrostOrigin::ensure_origin(origin)?;
+
+			let asset_a_loc = Location::try_from(*asset_a).map_err(|()| Error::<T>::BadVersion)?;
+			let asset_b_loc = Location::try_from(*asset_b).map_err(|()| Error::<T>::BadVersion)?;
+
+			let asset_a = T::CurrencyIdConvert::convert(asset_a_loc).ok_or(Error::<T>::AssetNotFound)?;
+			let asset_b = T::CurrencyIdConvert::convert(asset_b_loc).ok_or(Error::<T>::AssetNotFound)?;
+
+			let entry: OracleEntry<BlockNumberFor<T>> = {
+				let e = OracleEntry::new(
+					EmaPrice::new(price.0, price.1),
+					Volume::default(),
+					Liquidity::default(),
+					T::BlockNumberProvider::current_block_number(),
+				);
+				if ordered_pair(asset_a, asset_b) == (asset_a, asset_b) {
+					e
+				} else {
+					e.inverted()
+				}
+			};
+
+			Self::on_entry(BITFROST_SOURCE, (asset_a, asset_b), entry).map_err(|_| Error::<T>::TooManyUniqueEntries)?;
```

Comment:
here should be ordered pair, not (asset_a,asset_b)

Reply:
nice catch